python3Packages.libbs: init at 2.8.0

spencerpogo opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
Add the python library [libbs](https://github.com/binsync/libbs) and its dependencies. Motivation: `libbs` is a dependency of [angr-management](https://github.com/angr/angr-management) and this PR is a spinoff from #360310. 

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

spencerpogo committed:
python3Packages.pyhidra: init at 1.3.0

spencerpogo committed:
python3Packages.jfx-bridge: init at 1.0.0

spencerpogo committed:
python3Packages.ghidra-bridge: init at 1.0.0

spencerpogo committed:
python3Packages.libbs: init at 2.8.0

ethancedwards8 commented on pkgs/development/python-modules/libbs/default.nix:
```diff
@@ -0,0 +1,54 @@
+
+  meta = {
+    description = "Library for writing plugins in any decompiler: includes API lifting, common data formatting, and GUI abstraction";
+    homepage = "https://github.com/binsync/libbs";
```
add changelog

ethancedwards8 commented on pkgs/development/python-modules/jfx-bridge/default.nix:
```diff
@@ -0,0 +1,50 @@
+
+  meta = {
+    description = "Base Python2/3 RPC bridge used for ghidra_bridge";
+    homepage = "https://github.com/justfoxing/jfx_bridge";
```
add changelog

ethancedwards8 commented on pkgs/development/python-modules/ghidra-bridge/default.nix:
```diff
@@ -0,0 +1,39 @@
+
+  meta = {
+    description = "Python 3 bridge to Ghidra's Python scripting";
+    homepage = "https://github.com/justfoxing/ghidra_bridge";
```
add changelog

ethancedwards8 commented on pkgs/development/python-modules/ghidra-bridge/default.nix:
```diff
@@ -0,0 +1,39 @@
+  dependencies = [ jfx-bridge ];
+
+  pythonImportsCheck = [ "ghidra_bridge" ];
+
```
add `passthru.updateScript = nix-update-script { };`

ethancedwards8 commented on pkgs/development/python-modules/jfx-bridge/default.nix:
```diff
@@ -0,0 +1,50 @@
+  ];
+
+  pythonImportsCheck = [ "jfx_bridge" ];
+
```
add `passthru.updateScript = nix-update-script { };`

ethancedwards8 commented on pkgs/development/python-modules/libbs/default.nix:
```diff
@@ -0,0 +1,54 @@
+  ];
+
+  pythonImportsCheck = [ "libbs" ];
+
```
add `passthru.updateScript = nix-update-script { };`

ethancedwards8 commented on pkgs/development/python-modules/pyhidra/default.nix:
```diff
@@ -0,0 +1,35 @@
+  dependencies = [ jpype1 ];
+
+  pythonImportsCheck = [ "pyhidra" ];
+
```
add `passthru.updateScript = nix-update-script { };`

ethancedwards8 commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 369305`

---
### `aarch64-darwin`
<details>
  <summary>:x: 12 packages failed to build:</summary>
  <ul>
    <li>python312Packages.ghidra-bridge</li>
    <li>python312Packages.ghidra-bridge.dist</li>
    <li>python312Packages.jfx-bridge</li>
    <li>python312Packages.jfx-bridge.dist</li>
    <li>python312Packages.libbs</li>
    <li>python312Packages.libbs.dist</li>
    <li>python313Packages.ghidra-bridge</li>
    <li>python313Packages.ghidra-bridge.dist</li>
    <li>python313Packages.jfx-bridge</li>
    <li>python313Packages.jfx-bridge.dist</li>
    <li>python313Packages.libbs</li>
    <li>python313Packages.libbs.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.pyhidra</li>
    <li>python312Packages.pyhidra.dist</li>
    <li>python313Packages.pyhidra</li>
    <li>python313Packages.pyhidra.dist</li>
  </ul>
</details>

ethancedwards8 commented:
```
> FAILED jfx_bridge/test_bridge.py::TestBridge::test_remote_eval - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remote_eval_bad_code - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remote_eval_kwargs - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remote_eval_timeout - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remote_inheritance - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remote_iterable - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remote_iterable_for - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remoteify_class - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remoteify_class_with_inheritance - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remoteify_function_with_args - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remoteify_function_with_imports - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remoteify_function_with_kwargs - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remoteify_module - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remoteify_same_names - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remoteify_simple_function - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remoteify_unindented - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_serialize_deserialize_bytes - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_serialize_deserialize_types - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_set - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_slicing - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_truediv - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_unhashable - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_unicode_strings_only_when_required - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridgeZZZZZZZShutdown::test_zzzzzz_shutdown - PermissionError: [Errno 1] Operation not permitted
       > =========== 48 failed, 3 skipped, 10 deselected in 465.24s (0:07:45) ===========
       For full logs, run 'nix log /nix/store/6misscxz5j0faqfvix6wxmfrgz835vwg-python3.12-jfx-bridge-1.0.0.drv'.
error: 1 dependencies of derivation '/nix/store/2vngclx3ylvd5wg91vy5l19dnx1m10qb-python3.12-ghidra-bridge-1.0.0.drv' failed to build
error: 2 dependencies of derivation '/nix/store/zvyp61yicrz2gv297dxcbf9j0ypc97wl-python3.12-libbs-2.8.0.drv' failed to build
error: builder for '/nix/store/8rn37q0lfymvqz7sg46syjgmdqpm3j19-python3.13-jfx-bridge-1.0.0.drv' failed with exit code 1;
       last 25 log lines:
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remote_eval - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remote_eval_bad_code - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remote_eval_kwargs - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remote_eval_timeout - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remote_inheritance - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remote_iterable - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remote_iterable_for - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remoteify_class - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remoteify_class_with_inheritance - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remoteify_function_with_args - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remoteify_function_with_imports - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remoteify_function_with_kwargs - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remoteify_module - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remoteify_same_names - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remoteify_simple_function - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_remoteify_unindented - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_serialize_deserialize_bytes - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_serialize_deserialize_types - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_set - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_slicing - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_truediv - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_unhashable - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridge::test_unicode_strings_only_when_required - TimeoutError: timed out
       > FAILED jfx_bridge/test_bridge.py::TestBridgeZZZZZZZShutdown::test_zzzzzz_shutdown - PermissionError: [Errno 1] Operation not permitted
```

ethancedwards8 commented:
Looks like they're trying to get network access. Should probably disable just those specifically

spencerpogo commented:
Found it https://discourse.nixos.org/t/local-networking-in-checkphase/40208

fabaff committed:
python313Packages.jfx-bridge: refactor

fabaff committed:
python313Packages.ghidra-bridge: refactor

fabaff committed:
python313Packages.pyhidra: enable tests

fabaff committed:
python313Packages.libbs: enable tests

fabaff committed:
python313Packages.libbs: 2.8.0 -> 2.11.1

Diff: https://github.com/binsync/libbs/compare/refs/tags/v2.8.0...v2.11.1

Changelog: https://github.com/binsync/libbs/releases/tag/v2.11.1

fabaff commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 369305`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 16 packages built:</summary>
  <ul>
    <li>python312Packages.ghidra-bridge</li>
    <li>python312Packages.ghidra-bridge.dist</li>
    <li>python312Packages.jfx-bridge</li>
    <li>python312Packages.jfx-bridge.dist</li>
    <li>python312Packages.libbs</li>
    <li>python312Packages.libbs.dist</li>
    <li>python312Packages.pyhidra</li>
    <li>python312Packages.pyhidra.dist</li>
    <li>python313Packages.ghidra-bridge</li>
    <li>python313Packages.ghidra-bridge.dist</li>
    <li>python313Packages.jfx-bridge</li>
    <li>python313Packages.jfx-bridge.dist</li>
    <li>python313Packages.libbs</li>
    <li>python313Packages.libbs.dist</li>
    <li>python313Packages.pyhidra</li>
    <li>python313Packages.pyhidra.dist</li>
  </ul>
</details>
